mabwiser/softmax.py

```diff
@@ -63,14 +63,11 @@ def _expectation_operation(self):
 
         # Scaling range
         max_mean = max(self.arm_to_mean.values())
-        min_mean = min(self.arm_to_mean.values())
-        scaling_range = max_mean - min_mean
-        if scaling_range == 0:
-            scaling_range = 1
 
         # Scale the means and calculate the natural exponents --decrement max to avoid overflow from np.exp(x)
+        # Reference: https://stackoverflow.com/questions/42599498/numercially-stable-softmax
         for arm in self.arm_to_exponent:
-            self.arm_to_exponent[arm] = math.exp((self.arm_to_mean[arm] - max_mean) / (scaling_range * self.tau))
+            self.arm_to_exponent[arm] = math.exp((self.arm_to_mean[arm] - max_mean) / self.tau)
```

Comment:
cool.

From that reference, it seems SciPy also offers softmax. https://docs.scipy.org/doc/scipy/reference/generated/scipy.special.softmax.html#scipy.special.softmax 

It doesn't seem to have a "tau" parameter. 

Are we changing the formula in the docs as well?
https://fmr-llc.github.io/mabwiser/api.html#mabwiser.mab.LearningPolicy.Softmax

Reply:
Tau is from the Kuleshov multi-armed bandits paper. The formula is updated in the docs, it won't show on the website version until this pr is merged.

Reply:
This is the paper: https://arxiv.org/abs/1402.6028. It explains tau is used to control how random the choice is. This is the reinforcement learning variation on a softmax function.